File: apps/android/app/src/main/java/bot/molt/android/config/ThinkFleetConfig.kt
package bot.molt.android.config

object ThinkFleetConfig {
    /** Base URL for the SaaS API. Override via BuildConfig or system env. */
    val apiBaseUrl: String by lazy {
        System.getenv("THINKFLEET_API_URL")
            ?: try {
                val field = Class.forName("com.thinkfleet.android.BuildConfig").getField("THINKFLEET_API_URL")
                field.get(null) as? String ?: DEFAULT_URL
            } catch (_: Exception) {
                DEFAULT_URL
            }
    }

    private const val DEFAULT_URL = "https://www.thinkfleet.ai"

    const val SOCKET_IO_PATH = "/api/socket.io"
    const val RPC_BASE_PATH = "/api/rpc"
    const val AUTH_BASE_PATH = "/api/auth"

    const val PREFS_NAME = "thinkfleet_session"
}

File: apps/ios/Sources/Config/ThinkFleetConfig.swift
import Foundation

enum ThinkFleetConfig {
    /// Base URL for the SaaS API.
    /// Override via `THINKFLEET_API_URL` environment variable or Info.plist `ThinkFleetAPIURL` key.
    static let apiBaseURL: URL = {
        // 1. Environment variable (debug/CI)
        if let envURL = ProcessInfo.processInfo.environment["THINKFLEET_API_URL"],
           let url = URL(string: envURL) {
            return url
        }
        // 2. Info.plist (build configuration)
        if let plistURL = Bundle.main.object(forInfoDictionaryKey: "ThinkFleetAPIURL") as? String,
           !plistURL.isEmpty,
           let url = URL(string: plistURL) {
            return url
        }
        // 3. Default — production
        return URL(string: "https://www.thinkfleet.ai")!
    }()

    static let socketIOPath = "/api/socket.io"
    static let rpcBasePath = "/api/rpc"
    static let authBasePath = "/api/auth"

    static let keychainService = "ai.thinkfleet.thinkfleetAI"
    static let appGroupIdentifier = "group.ai.thinkfleet.thinkfleetAI"
}
